src/ol/interaction/Draw.js

```diff
@@ -333,27 +333,25 @@ class Draw extends PointerInteraction {
            * +-------+------<3>-- (the third point may be anywhere on this line)
            */
 
-          if (coordinates.length > 2) {
+           if (coordinates.length > 2) {
             var first = coordinates[0];
             var second = coordinates[1];
             var third = coordinates[2];
 
             // vector from first to second
             var a_vec = [second[0] - first[0], second[1] - first[1]];
 
-            if (a_vec[1] === 0) {
+            if (a_vec[0] === 0 && a_vec[1] === 0) {
                 // catch the case where the first and second point are equal
-                coordinates = [[first, first, first, first]];
+                coordinates = [[first]];
             } else {
               // perpendicular vector to a_vec
               var b_vec = [-1 * a_vec[1], a_vec[0]];
 
-              // helper
-              var tmp = a_vec[0] / a_vec[1];
               // compute the intersection parameter of the two lines
               // going from second in b_vec direction
               // and from third in a_vec direction
-              var x = (third[0] + tmp * (second[1] - third[1]) - second[0]) / (b_vec[0] - b_vec[1] * tmp);
+              var x = (third[0] * second[1] - third[1] * second[0]) / (third[0] * b_vec[1] - third[1] * b_vec[0]);
```

Comment:
This would now crash if `third` was (0, 0). I tried to do this but I couldn't, so maybe we can ignore this case here.

Reply:
Somehow it's not possible to get (0,0) as third point any more. But I don't know how to rewrite the formula without using `b_vec` and `third` in the denominator. Should we still leave it like this? Another idea would be to add a little bias (e.g. +0.0001), if (0,0) is reached.

Reply:
The idea with adding a small value is good! I was about to suggest adding a value all the time but then it could fail if `third` is -0.0001 and the value is +0.0001. I didn't think of a simple check if `third` is 0 :smile: 

I recently saw a similar case where they used `1e-7` as value, so let's randomly use that, too.